python/nvtx/nvtx.py

```diff
@@ -64,12 +64,12 @@ def __init__(self, message=None, color=None, domain=None, category=None):
         """
 
         self.domain = Domain(domain)
-
-        category_id = (
-            self.domain.get_category_id(category)
-            if category is not None
-            else None
-        )
+ 
+        category_id = None
+        if isinstance(category, int):
+            category_id =  category
+        elif isinstance(category, str):
+            category_id = self.domain.get_category_id(category)
```

Comment:
This logic seems to be repeated in a few different places -- should we just make it part of `get_category_id`?